### ISSUE TITLE
don't store timezone data in repo?

### ISSUE THREAD

**Issue by elliefm** [cyrusimap/cyrus-timezones#7]
This currently stores a copy of the IANA data in the repo, which might be out of date by now

It would probably be more useful/need less handholding if it had a make rule that would fetch and extract the data direct from IANA, rather than expecting it to be kept up to date in the repo

**Comment by shamefulCake1** [comment]
Cannot this package just use glibc-zoneinfo?

**Comment by rsto** [comment]
> This currently stores a copy of the IANA data in the repo, which might be out of date by now

It is not, the current IANA timezone database version is `2025b`.

> It would probably be more useful/need less handholding if it had a make rule that would fetch and extract the data direct from IANA, rather than expecting it to be kept up to date in the repo

Sounds good to me.

> Cannot this package just use glibc-zoneinfo?

cyrus-timezones contains IANA timezones represented as VTIMEZONEs, which Cyrus requires for CalDAV and JMAP Calendars support. It also contains a library that allows to guess an equivalent IANA timezone for an arbitrary timezone definition. Neither seems to be provided by glibc-zoneinfo?

**Comment by shamefulCake1** [comment]
> cyrus-timezones contains IANA timezones represented as VTIMEZONEs
>Neither seems to be provided by glibc-zoneinfo?

glibc-zoneinfo provides information about IANA timezones (I am not sure whether in the VTIMEZONE format). 
I guess it would make sense to just use that data? tzdata is usually updated by the system vendors on a regular basis.

https://mirrors.slackware.com/slackware/slackware64-current/source/a/glibc-zoneinfo/ contains tzdata2025b.

>It also contains a library that allows to guess an equivalent IANA timezone for an arbitrary timezone definition.

I do not think it does, no. But aren't these two tasks (providing tz data and guessing timezones) mostly unrelated?

>a make rule that would fetch and extract the data direct from IANA

Downloading data from a makefile is a horrible practice, since make is often run as `root` to make a package. Moreover, people usually download `tar.gz` packages from github expecting them to be self-contained, and are very frustrated when it turns out that those packages are unusable without public internet access.

**Comment by rsto** [comment]
> glibc-zoneinfo provides information about IANA timezones (I am not sure whether in the VTIMEZONE format). I guess it would make sense to just use that data?

Agreed, at least it would make sense to allow setting the zoneinfo directory in the Makefile from which the VTIMEZONEs are built.

We still need cyrus-timezones though, it came into existence because libical packages ship their own VTIMEZONE definitions, and those typically are not updated for every new timezone definition.

> https://mirrors.slackware.com/slackware/slackware64-current/source/a/glibc-zoneinfo/ contains tzdata2025b.
> 
> > It also contains a library that allows to guess an equivalent IANA timezone for an arbitrary timezone definition.
> 
> I do not think it does, no. But aren't these two tasks (providing tz data and guessing timezones) mostly unrelated?

Yes, bundling them in one repo is more a matter of convenience. The guesstz library could just as well be unbundled and its Makefile be tweaked to use a custom zoneinfo dir.

All that being said: you don't have to use cyrus-timezones at all to build Cyrus. If you don't, you need to make sure that your VTIMEZONE definitions, e.g. those shipped with libical, are up to date and the `zoneinfo_dir` imapd.conf option points to them. You would lose guessing the IANA equivalent for custom timezones, but luckily that's only required for a very small fraction of iCalendar.

**Comment by rsto** [comment]
This repository now is archived.
